include/seqan3/core/detail/type_inspection.hpp

```diff
@@ -41,18 +41,28 @@ namespace seqan3::detail
  * \note The returned name is implementation defined and might change between different tool chains.
  */
 template <typename type>
-inline static const std::string type_name_as_string = [] ()
+inline std::string const type_name_as_string = [] ()
 {
+    std::string ret;
```

Comment:
```suggestion
    std::string demangled_name{};
```

Reply:
I used `ret` throughout the library for the return value of lambda expressions, especially in the context of immediately evaluated lambdas used in the initialisation of variables.
But I can change it here, it's not important.